src/main/webapp/app/file-upload-assessment/file-upload-assessment.component.ts

```diff
@@ -0,0 +1,482 @@
+                // Set min height
+                modifiers: [
+                    interact.modifiers!.restrictSize({
+                        min: { width: 0, height: this.resizableMinHeight },
+                    }),
+                ],
+                inertia: true,
+            })
+            .on('resizestart', function(event: any) {
+                event.target.classList.add('card-resizable');
+            })
+            .on('resizeend', function(event: any) {
+                event.target.classList.remove('card-resizable');
+            })
+            .on('resizemove', function(event) {
+                const target = event.target;
+                // Update element height
+                target.style.minHeight = event.rect.height + 'px';
+                $('#submission-area').css('min-height', event.rect.height - 100 + 'px');
+            });
+    }
+
+    public ngOnDestroy(): void {
+        this.changeDetectorRef.detach();
+    }
+
+    public addReferencedFeedback(): void {
+        const referencedFeedback = new Feedback();
+        referencedFeedback.credits = 0;
+        this.referencedFeedback.push(referencedFeedback);
+        this.validateAssessment();
+    }
+
+    public deleteAssessment(assessmentToDelete: Feedback): void {
+        const indexToDelete = this.referencedFeedback.indexOf(assessmentToDelete);
+        this.referencedFeedback.splice(indexToDelete, 1);
+        this.validateAssessment();
+    }
+
+    /**
+     * Load next assessment in the same page.
+     * It calls the api to load the new unassessed submission in the same page.
+     * For the new submission to appear on the same page, the url has to be reloaded.
+     */
+    assessNextOptimal() {
+        if (this.exercise.type === ExerciseType.FILE_UPLOAD) {
+            this.fileUploadSubmissionService.getFileUploadSubmissionForExerciseWithoutAssessment(this.exercise.id).subscribe(
+                (response: FileUploadSubmission) => {
+                    this.unassessedSubmission = response;
+                    this.router.onSameUrlNavigation = 'reload';
+                    // navigate to the new assessment page to trigger re-initialization of the components
+                    this.router.navigateByUrl(`/file-upload-exercise/${this.exercise.id}/submission/${this.unassessedSubmission.id}/assessment`, {});
+                },
+                (error: HttpErrorResponse) => {
+                    if (error.status === 404) {
+                        // there are no unassessed submission, nothing we have to worry about
+                        this.jhiAlertService.error('artemisApp.tutorExerciseDashboard.noSubmissions');
+                    } else {
+                        this.onError(error.message);
+                    }
+                },
+            );
+        }
+    }
+
+    onSaveAssessment() {
+        this.fileUploadAssessmentsService.saveAssessment(this.assessments, this.submission!.id).subscribe(
```

Comment:
Something I wonder in general here: There doesn't seem to be any user feedback while a REST call is pending. E.g. what if saveAssessment would take 10 seconds? Would the user understand the process?

Reply:
You are right that this was missing, I set `isLoading` flag here